### DISCUSSION TITLE
GraphQL pagination

### DISCUSSION THREAD

**Discussion by ShreyasSurendra** [netbox-community/netbox#11322] [Q&A]
Does NetBox support graphql pagination? If yes can somebody please share an example of a query with pagination.

**Comment by tyler-8** [comment]
I don't think it does today, but it would be a huge benefit to the GraphQL APIs.

#Details for research:

## Documentation

- https://docs.graphene-python.org/projects/django/en/latest/queries/#relay
- https://graphql.org/learn/pagination/

## Related `django-graphene` issues

- https://github.com/graphql-python/graphene-django/issues/809
- https://github.com/graphql-python/graphene-django/issues/751
- https://github.com/graphql-python/graphene-django/issues/829

-------------------------

A cursor-based implementation would be ideal and the syntax would look something like:

```
{
    site_list(first: 2, offset:6) {
        name
        status
    }
}
```

This would retrieve the first 2 sites in the list, starting with the 7th entry in the list using the default sorting.
